### PR TITLE
shrinking the deprecated span

### PR DESCRIPTION
**Pull request by 0xPoe**
ref: https://github.com/rust-lang/rust/pull/85617#issuecomment-854947988

part of #85403

r? @estebank

The reason is that if we use method_span directly, it will cause the in_derive_expansion judgment to fail.

### CODE REVIEW

File: compiler/rustc_middle/src/middle/stability.rs [hunk 1]
```diff
@@ -226,18 +226,19 @@ fn late_report_deprecation(
     suggestion: Option<Symbol>,
     lint: &'static Lint,
     span: Span,
+    method_span: Option<Span>,
```
Comment (hi-rustin) [inline compiler/rustc_middle/src/middle/stability.rs:229]: Maybe it's not a relevant name for now. But I'm not quite sure what to rename it.

**Comment by hi-rustin** [comment]
@estebank Ping~  Could you please take a look? I would be very grateful.

File: compiler/rustc_middle/src/middle/stability.rs (hunk 1 above)
Comment (estebank) [reply compiler/rustc_middle/src/middle/stability.rs:229]: Same.

**Pull request merged**